RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticCluster.h

```diff
@@ -0,0 +1,85 @@
+#ifndef __RecoParticleFlow_PFClusterProducer_RealisticCluster_H__
+#define __RecoParticleFlow_PFClusterProducer_RealisticCluster_H__
+
+#include <array>
+#include <vector>
+#include <algorithm>
+
+class RealisticCluster
+{
+
+    public:
+
+        RealisticCluster()
+        {
+            totalEnergy = 0.f;
+            exclusiveEnergy = 0.f;
+            visible = true;
+        }
+        void increaseEnergy(float value)
+        {
+            totalEnergy+=value;
+        }
+        void increaseExclusiveEnergy(float value)
+        {
+            exclusiveEnergy+=value;
+        }
+
+        float getExclusiveEnergyFraction() const
+        {
+            float fraction = 0.f;
+            if(totalEnergy>0.f){
+                fraction = exclusiveEnergy/totalEnergy;
+            }
+            return fraction;
+        }
+
+        float getEnergy() const
+        {
+            return totalEnergy;
+        }
+
+        float getExclusiveEnergy() const
+        {
+            return exclusiveEnergy;
+        }
+
+        bool isVisible() const
+        {
+            return visible;
+        }
+
+        void setVisible(bool vis)
+        {
+            visible = vis;
+        }
+
+        void addHitAndFraction(unsigned int hit, float fraction)
+        {
+            hitIdsAndFractions_.emplace_back(hit,fraction);
+        }
+//TODO: hits are sorted, replace with binary search.
```

Comment:
Are the hits already sorted in this PR? If so, `find_if` could just be replaced with `lower_bound`